vtex/toolbelt
# Fix payload too large

#### What is the purpose of this pull request?
Fix PayloadTooLarge error by:
- Truncate strings when necessary
- Set encoding-type in request
- Use a buffer for request body

#### What problem is this solving?
PayloadTooLarge error.

#### How should this be manually tested?
Go to account `storecomponents`, workspace `toolbelttest` where the [necessary version](https://github.com/vtex/toolbelt-telemetry/pull/6) of `toolbelt-telemetry` is linked. 

Then clone `toolbelt` and run `yarn watch` in this branch with the commands bellow:
```
cd /tmp && rm -rf toolbelt && \
git clone https://github.com/vtex/toolbelt.git && \
cd toolbelt && \
git checkout fix/deprecation-check && \
yarn && yarn watch
```

Throw an error with a large message in a command file. For example, in `src/modules/apps/list.ts` in beginning of the default export put the line:
```
throw new Error('a'.repeat(2 ** 20))
```

The use:
```
vtex-test ls
```
To run the command that will thrown the error. After some time, your error should appear in splunk. (Try the query: `index=io_vtex_logs app="vtex.toolbelt-telemetry@*" level=error workspace=toolbelttest`)

#### Screenshots or example usage

#### Types of changes

#### Chores checklist
